enforcer_synced.go

```diff
@@ -53,10 +53,10 @@ func (e *SyncedEnforcer) IsAutoLoadingRunning() bool {
 // StartAutoLoadPolicy starts a go routine that will every specified duration call LoadPolicy
 func (e *SyncedEnforcer) StartAutoLoadPolicy(d time.Duration) {
 	// Don't start another goroutine if there is already one running
-	if e.IsAutoLoadingRunning() {
+	if !atomic.CompareAndSwapInt32(&e.autoLoadRunning, 0, 1) {
```

Comment:
If the e.IsAutoLoadingRunning method is not safe, we should fix it

Reply:
method is safe.
but mechanism that started goroutine, consist of two parts:
 - check if goroutine already started
 - set flag `autoLoadRunning`

two or more coroutines can execute this steps at some time for each step

Reply:
@sgoroshko Ok

Reply:
@sgoroshko

Why not directly change the function body of `IsAutoLoadingRunning()`?

Reply:
@hsluoyz `IsAutoLoadingRunning()` just check value, `atomic.CompareAndSwapInt32` can update the `e. autoLoadRunning` value.

Reply:
OK